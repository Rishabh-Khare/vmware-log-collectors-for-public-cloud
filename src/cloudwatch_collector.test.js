const lintEnv = require('./lint_env').lintTestEnv;
const { createSample1 } = require('./cloudwatch_testdata');
const { gzipLogs } = require('./lint');
const nock = require('nock');

const {
  sendLogs,
  CloudWatchHttpCollector,
  CloudWatchKafkaCollector,
} = require('./index');

describe('CloudWatchKafkaCollector', () => {
  const collector = new CloudWatchKafkaCollector(lintEnv);

  it('processLogsJson', () => {
    const logsJson = createSample1();
    collector.processLogsJson(logsJson);
    expect(logsJson.subscriptionFilters).toBe(undefined);
    expect(logsJson.logEvents).toBe(undefined);
    expect(logsJson).toMatchSnapshot();
  });
});

describe('CloudWatchHttpCollector', () => {
  const collector = new CloudWatchHttpCollector(lintEnv);

  describe('processLogsJson', () => {
    it('should process logs JSON', () => {
      const logsJson = createSample1();
      collector.processLogsJson(logsJson);
      expect(logsJson).toMatchSnapshot();
    });
  });

  describe('sendLogs', () => {
    const sendData = (done) => {
      nock('https://data.mock.symphony.com')
        .post('/le-mans/v1/streams/ingestion-pipeline-stream')
        .reply(200, { message: 'ok' });

      const logsJson = createSample1();
      gzipLogs(logsJson)
        .then(zippedData => sendLogs(zippedData, collector))
        .then(() => done())
        .catch(error => done.fail(error));
    };

    it('should send request to the HTTP stream', (done) => {
      sendData(done);
    });

    // it('should send request to the Kafka stream', (done) => {
    //   sendData(false, done);
    // }, 40000);
  });
});
