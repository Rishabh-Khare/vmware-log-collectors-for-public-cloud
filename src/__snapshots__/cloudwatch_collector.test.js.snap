// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudWatchHttpCollector processLogsJson should process logs JSON 1`] = `
Object {
  "logEvents": Array [
    Object {
      "id": "34051922485820161636472577410016310394916410432944603138",
      "text": "2018-05-21T22:16:53.934Z	aab15e49-5d44-11e8-a7db-afa704cec35a	Status: 200
",
    },
    Object {
      "id": "001",
      "message": "message of 001",
      "text": "text of 001",
    },
    Object {
      "id": "34051922485820161636472577410016310394916410432944603138",
      "text": Array [
        Object {
          "fielda": "fielda",
        },
      ],
    },
  ],
  "logGroup": "/aws/lambda/runbook",
  "logStream": "unittest8",
  "messageType": "DATA_MESSAGE",
  "owner": "792677813823",
}
`;

exports[`CloudWatchKafkaCollector processLogsJson 1`] = `
Object {
  "logGroup": "/aws/lambda/runbook",
  "logStream": "unittest8",
  "logs": Array [
    Object {
      "id": "34051922485820161636472577410016310394916410432944603138",
      "text": "2018-05-21T22:16:53.934Z	aab15e49-5d44-11e8-a7db-afa704cec35a	Status: 200
",
    },
    Object {
      "id": "001",
      "message": "message of 001",
      "text": "text of 001",
    },
    Object {
      "id": "34051922485820161636472577410016310394916410432944603138",
      "text": Array [
        Object {
          "fielda": "fielda",
        },
      ],
    },
  ],
  "messageType": "DATA_MESSAGE",
  "owner": "792677813823",
  "structure": "cloudwatch",
}
`;
